Assets/RGLUnityPlugin/Scripts/SceneManager.cs

```diff
@@ -409,5 +410,41 @@ private static bool IsNotActiveOrParentHasLidar(GameObject gameObject)
         {
             return !gameObject.activeInHierarchy || gameObject.GetComponentsInParent<LidarSensor>().Length != 0;
         }
+
+        private static bool TryCreateRGLObject<T>(T meshSource, out IRGLObject rglObject) where T : UnityEngine.Object
+        {
+            try
+            {
+                if (meshSource is MeshRenderer mr)
+                {
+                    rglObject = new RGLMeshRendererObject(mr);
+                }
+                else if (meshSource is SkinnedMeshRenderer smr)
+                {
+                    rglObject = new RGLSkinnedMeshRendererObject(smr);
+                }
+                else if (meshSource is Collider collider)
+                {
+                    rglObject = new RGLColliderObject(collider);
+                }
+                else if (meshSource is Terrain terrain)
+                {
+                    rglObject = new RGLTerrainObject(terrain);
+                }
+                else
+                {
+                    Debug.LogError($"Could not create RGLObject from type '{typeof(T)}'");
```

Comment:
Logically:
Why is one case a warning only, but the second one is an error?

Reply:
The warning occurs when the RGLObject constructor fails - the type of the GameObject is correct, but its configuration is invalid.
The error occurs when RGLObject cannot be created from the GameObject because of the type. This indicates a programming bug - that function shouldn't be called on that GameObject.